test/Streamly/System/Process.hs

```diff
@@ -448,6 +449,10 @@ processChunks'4 = monadicIO $ run checkFailAction
 
 main :: IO ()
 main = do
+    -- Make sure all the executable files have the executable bit set before
+    -- running the tests.
+    for_ [executableFile, executableFileFail, executableFilePass]
+        $ \f -> S.drain $ Proc.toChunks "chmod" ["+x", f]
```

Comment:
We should not be making changes to the source tree for running tests. We can just remove the execute permissions on these files and run them using the shell like `sh <filename` or `/usr/bin/env sh <filename>`.